src/core/xfa/template.js

```diff
@@ -5288,13 +5288,17 @@ class Template extends XFAObject {
           } else if (target instanceof ContentArea) {
             const index = contentAreas.findIndex(e => e === target);
             if (index !== -1) {
+              // In the next loop iteration `i` will be incremented, note the
+              // `continue` just below, hence we need to subtract one here.
               i = index - 1;
             } else {
               targetPageArea = target[$getParent]();
-              startIndex =
-                targetPageArea.contentArea.children.findIndex(
-                  e => e === target
-                ) - 1;
+              const childIndex = targetPageArea.contentArea.children.findIndex(
+                e => e === target
+              );
+              if (childIndex !== -1) {
```

Comment:
This check should be useless: the target (contentArea) is necessarly a children of its parent.

Reply:
And in case it's possible to have -1 I'd prefer having an error, but I'm sure it can't.

Reply:
> This check should be useless: the target (contentArea) is necessarly a child of its parent.

The reason for this added check, based on no actual knowledge of either this code or the XFA specification was:
 - To prevent issues if the `children` Array would ever be empty; although I'm not sure if that can ever happen for real-life XFA documents (e.g. corrupt ones)?
 - To prevent issues if the specific `target` cannot be found in the `children` Array; again I'm not sure if that can happen for real-life XFA documents (e.g. corrupt ones)?

If neither of those error-case can ever happen, I'll just remove it this check; otherwise throwing a `FormatError` is probably appropriate.
@calixteman Please let me know what you prefer, thank you!

Reply:
A `contentArea` can only be under a `pageArea`. I mean it isn't possible to have in the JS representation of a node something which isn't defined in the constructor, so even with ill-formatted xml it's impossible to have a `contentArea` not under a `pageArea`. The constructor of each class in `template.js` must be the exact reflection of the xml node as we have in the specs: so normally, I mean modulo bugs, the xml is validated. 
So in this case the `$getParent` will always return a `pageArea` and this one must contain one of its child (i.e. the `contentArea`). The parent is attached to its child when the child is accepted by its future parent:
https://github.com/mozilla/pdf.js/blob/6c9b6bc59999cfe7253ecc1323f38210b8739cd0/src/core/xfa/xfa_object.js#L132
So having a -1 here means that the parent is not really the parent and something is really wrong in the code but not in the pdf.
So I'd just remove the check but if you feel that's better to throw an exception in this case I'm fine with that too.

Reply:
Thank you for the in-depth explanation!

> So having a -1 here means that the parent is not really the parent and something is really wrong in the code but not in the pdf.

Given that this suggests a bug in our code, and not simply a corrupt XFA document, I'm completely fine with just setting the `startIndex` and not worry about the `-1` case. I'll update the patch accordingly, thanks!